stl/inc/functional

```diff
@@ -29,6 +29,17 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
+#if _HAS_CXX23
+// clang-format off
+template <class _Result_type, class _Callable, class _Ty1, class... _Types2>
+requires is_invocable_r_v<_Result_type, _Callable, _Ty1, _Types2...>
+constexpr _Result_type invoke_r(_Callable&& _Obj, _Ty1&& _Arg1, _Types2&&... _Args2)
+noexcept(is_nothrow_invocable_r_v<_Result_type, _Callable, _Ty1, _Types2...>) {
+    return _STD invoke(_STD forward<_Callable>(_Obj), _STD forward<_Ty1>(_Arg1), _STD forward<_Types2>(_Args2)...);
+}

```

Comment:
```suggestion
template <class _Result_type, class _Callable, class... _Types>
    requires is_invocable_r_v<_Result_type, _Callable, _Types...>
constexpr _Result_type invoke_r(_Callable&& _Obj, _Types&&... _Args) noexcept(
    is_nothrow_invocable_r_v<_Result_type, _Callable, _Types...>) {
    if constexpr (is_void_v<_Result_type>) {
        (void) _STD invoke(_STD forward<_Callable>(_Obj), _STD forward<_Types>(_Args)...);
    } else {
        return _STD invoke(_STD forward<_Callable>(_Obj), _STD forward<_Types>(_Args)...);
    }
}
```
Ignoring return value by `invoke_r<void>(...)` and calling with zero argument (e.g. `std::invoke_r<void>([]{ return 0; })`) are needed to be supported.